protocol/check.go

```diff
@@ -0,0 +1,52 @@
+package protocol
+
+import (
+	"bytes"
+	"context"
+	"errors"
+
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+// expectedBytecode is generated by deploying an instance of SwapFactory.sol and calling `CodeAt`
```

Comment:
I was having trouble replicating the `expectedByteCode`, because you didn't mention what address it was deployed with.  Parsing it out, it looks like it was deployed with `0x79d5a6f7d950e2b616f638f670bdb124bb9b6b67`.